### PR TITLE
Web Mercator support

### PR DESCRIPTION
**Pull request by procrastinatio**
- [x] Search
- [x] Find
- [x] Identify

### CODE REVIEW

**Comment by loicgasser** [comment]
great! can you make the pull request in the mvt branch please?

**Comment by davidoesch** [comment]
My personal Xmas wish: a print solution

**Comment by procrastinatio** [comment]
@davidoesch 
So, I guess the same print solution for 2D, 3D, raster and vector?
Client-side printing is more or less a glorified print screen.
Is support/control on print resolution/scale/pdf version, etc. important ? Yes, backend (headless browser), no, client-side.

If you like blah blah, https://github.com/procrastinatio/web-map-printing/blob/master/web-map-printing.pdf

**Comment by davidoesch** [comment]
resolution/scale/pdf version Is the key. And what it takes to do this... Your document actually gives a rough estimate of the work needed

**Comment by procrastinatio** [comment]
@davidoesch  
Ye old print works with any flat earth projections (don't care as long as long it is cartesian), WMS, WMTS and KML

https://mf-chsdi3.int.bgdi.ch/mvt/shorten/7e7c3103b4  (fork from https://github.com/geoadmin/mf-geoadmin3/pull/4523) with print enabled.

Now, if you want to print the vector layer, some possibilities.
- Rasterize them with tileserver-gl, basically a `TOD` for vector (big project)
- Implement vector tile as source in mapfish-print, which is already a dead project (not recommended, not very WYSIWIG as support for vector in MFPv2 is very basic)
- Pass the vector tile  the same way it is done for KML (will work only for layers of less than a few MB)
- Use map.geo.admin.ch to rasterize the vector layer as a big image (no control on quality) and send it someway to the print server.
- etc.

**Comment by davidoesch** [comment]
The PDF with vector tiles does not look to bad... From the options you gave: well I smells like a huge effort- like MVT itself-Needs to be taken into a separate project

**Comment by oterral** [comment]
@procrastinatio can you fix the rebase on mvt branch ?

**Comment by procrastinatio** [comment]
@ltclm The release service returns a different list between LV03 and WGS1984. These years are not in LV03:
- [u'18961231',
-  u'19051231',
-  u'18891231',
-  u'19121231',
-  u'19281231',
-  u'19161231',
-  u'19661231',
-  u'19751231',
-  u'19201231']

**Comment by procrastinatio** [comment]
@oterral We'll merge next week

**Comment by ltclm** [comment]
is it planned to merge the web mercator stuff into the master?

**Comment by danduk82** [comment]
jenkins test

File: chsdi/views/search.py [hunk 1]
```diff
@@ -393,11 +394,34 @@ def _add_feature_queries(self, queryText, timeFilter):
...
+
         if not self.returnGeometry:
             attrs2Del = ['x', 'y', 'lon', 'lat', 'geom_st_box2d']
             popAtrrs = lambda x: res.pop(x) if x in res else x
             map(popAtrrs, attrs2Del)
+        elif int(self.srid) not in (21781, 2056):
+            self._box2d_transform(res)
+            try:
+                pnt = (res['y'], res['x'])
+                x, y = transform_coordinate(pnt, self.DEFAULT_SRID, self.srid)
```
Comment (danduk82) [inline chsdi/views/search.py]: in this case, if we are in `4326` we should use directly what is coming from sphinx as `lat` and `lon` (if exist)

prio 2

File: tests/integration/__init__.py [hunk 2]
```diff
@@ -82,3 +105,6 @@ def assertBBoxValidity(self, bbox, srid):
         if srid == 2056:
             self.assertGreater(bbox[0], self.grids['2056'].MINX)
             self.assertGreater(bbox[1], self.grids['2056'].MINY)
+        if srid in (3857, 4326):
```
Comment (danduk82) [inline tests/integration/__init__.py]: as discussed please refactor and check this test (does it work at all?)

```python
if srid in SUPPORTED_OUTPUT_SRS:
  etc
```

File: tests/integration/test_identify_service.py [hunk 3]
```diff
@@ -270,7 +270,7 @@ def test_identify_with_geojson(self):
         self.assertEqual(resp_1.json['results'][0]['id'], resp_2.json['results'][0]['id'])
 
     def test_identify_with_geojson_returned_geometry(self):
-        params = {'geometry': '600000,200000,631000,210000',
+        params = {'geometry': '600000,200000,6020000,2020000',  # 600000,200000,631000,210000',
```
Comment (danduk82) [inline tests/integration/test_identify_service.py]: please cleanup

File: chsdi/models/vector/__init__.py [hunk 4]
```diff
@@ -100,7 +100,7 @@ def __read__(self):
     def srid(self):
         return self.geometry_column().type.srid
 
-    def to_esrijson(self, trans, returnGeometry):
+    def to_esrijson(self, trans, returnGeometry, out_srid):
```
[the remark this answers is not in the document]
Comment (danduk82) [reply chsdi/models/vector/__init__.py]: hi @procrastinatio , the `out_srid` is still present in `chsdi/models/vector/__init__.py` in functions `to_esrijson`and `to_geojson`, or do I have some strange status in my github view?

File: chsdi/views/search.py [hunk 5]
```diff
@@ -415,13 +415,20 @@ def _parse_locations(self, res):
...
-                res['x'] = x
-                res['y'] = y
-            except Exception:
-                raise exc.HTTPInternalServerError('Error while converting point(x, y) to EPSG:{}'.format(self.srid))
+            if int(self.srid) == 4326:
+                try:
+                    res['x'] = res['lon']
+                    res['y'] = res['lat']
+                except KeyError:
+                    raise exc.HTTPInternalServerError('Sphinx location has no lat/long defined')
```
Comment (danduk82) [inline chsdi/views/search.py]: @ltclm , do you know if in sphinx response lat/lon are **always** defined?

**Comment by procrastinatio** [comment]
@danduk changes done

File: chsdi/views/search.py (hunk 5 above)
Comment (ltclm) [reply chsdi/views/search.py]: yes, they're always defined.

**Pull request merged**